server/src/main/java/org/elasticsearch/internal/VersionExtension.java

```diff
@@ -24,6 +27,13 @@ public interface VersionExtension {
      */
     TransportVersion getCurrentTransportVersion(TransportVersion fallback);
 
+    /**
+     * Returns collection of {@link TransportVersion} defined by extension
+     */
+    default Collection<TransportVersion> getTransportVersions() {
```

Comment:
I think `getAllTransportVersions()` might be slightly misleading, because the method doesn't return all known transport versions, it returns transport versions defined by extension, so I would prefer to keep name as `getTransportVersions()`.

I removed a default implementation